.github/workflows/build.yml

```diff
@@ -99,26 +111,34 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python3: ${{fromJson(needs.common_config.outputs.linux_python_versions)}}
-        python_deps: [""]
+        python3: ${{fromJson(vars.LINUX_PYTHON_VERSIONS || '[6, 7, 8, 9, 10, 11]')}}
         include:
+          - python_deps_ids: [""]
+            matrix_override: ${{fromJson(needs.common_config.outputs.linux_matrix)}}
           - python3: 6
-            python_deps: "requirements-compatibility-py36.txt"
+            python_deps_ids: ["", -compat36]
+            matrix_override:
+              - ${{fromJson(needs.common_config.outputs.linux_matrix)[0]}}
+              - python_deps_id: -compat36
+                python_deps: requirements-compatibility-py36.txt
           - python3: 8
-            python_deps: "requirements-compatibility-py38.txt"
-    name: 3.${{matrix.python3}} Linux ${{matrix.python_deps}}
+            python_deps_ids: ["", -compat38]
```

Comment:
Why the "" first element? I would have thought the first element in the `include` would take care of this, and no need for python_deps_ids to be a string-JSON-list?

Reply:
The way `matrix.include` is expanded is like this: 

For each map object in the `include` list:
1. if any map entry matches the existing case(s), and no entry conflict with such case(s), merge the entire map into all of those cases
2. if there were conflict(s), the map becomes its own separate case (no inheritance from any existing case)
3. if there is no overlap at all, merge the map into all existing cases
4. except if there's no existing case, then the map becomes the first case.

What I need to do in `build_steps.yml::python_tests` matrix is to: declare two `python_deps_ids` case and apply the `inputs.matrix` to both of them.

I have tried the following:
* In the `linux_matrix` of `common_config` above, add two `- python_deps_ids: ...` items before the current `- os: linux` entry. This doesn't work because there's no easy way in GitHub Actions (GHA) to address only the last entry for the `build_steps.yml::compile` matrix
* Split `common_config` into named parts like "common_matrix", "text_matrix_extra". This is difficult because GHA doesn't have a native way to join an item ("common_matrix") with a list ("text_matrix_extra"). The hack to do this will be very difficult to understand

This led me to this solution:
1. Declare the `python_deps_ids` so they create some existing cases in the matrix
2. Make sure the `- os: linux` is the first item so it can be easily selected for the `compile` job. For the `python_tests` job, as it has no keys that overlap with existing matrix cases, it will be merged into _all_ cases
3. Add extra `- python_deps_ids: -compat...` item in the `matrix_override` which will be merged into one of the existing cases